CaImAn/caiman_main.py

```diff
@@ -0,0 +1,282 @@
+# Import miscellaneous and utilities librarys
+import os
+import sys
+import h5py
+import cv2
+import numpy as np
+import logging
+from typing import Any, Dict, List, Optional, Tuple, Union, Callable
+
+sys.path.append("..")
+# Import CaimAn related utilities libraries
+import utilities as utils
+import definitions as defs
+import caiman_definitions as cm_defs
+import caiman_parameters  as cm_params
+
+# Import for CaimAn lib
+import caiman as cm
+
+# Import for PyInstaller
+from multiprocessing import freeze_support
+freeze_support()
+
+
+def main(caiman_parameters):
+    """
+    CaImAn CNMF algorithm
+    """
+
+    #os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+
+    try:
+        cv2.setNumThreads(0)
+    except:
+        pass
+
+    # start the cluster
+    try:
+        cm.stop_server()  # stop it if it was running
+    except():
+        pass
+
+    c, dview, n_processes = cm.cluster.setup_cluster(backend="local", n_processes=None, single_thread=False)
+
+    fname_new = motion_correction(dview, caiman_parameters)
+
+    Yr, dims, T, cnm, images = cnmf(n_processes, dview, fname_new, caiman_parameters)
+
+    # Save results to .doric file
+    print(cm_defs.Messages.SAVING_DATA, flush=True)
+
+    file_ = h5py.File(caiman_parameters.paths[defs.Parameters.Path.FILEPATH], 'r')
+    # Get all operation parameters and dataset attributes
+    data, driver, operation, series, sensor = caiman_parameters.get_h5path_names()
+    params_source_data = utils.load_attributes(file_, f"{data}/{driver}/{operation}")
+
+    attrs = utils.load_attributes(file_, f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/{caiman_parameters.dataname}")
+    time_ = np.array(file_[f"{caiman_parameters.paths[defs.Parameters.Path.H5PATH]}/{defs.DoricFile.Dataset.TIME}"])
+
+    file_.close()
+
+    Y = np.transpose(images, list(range(1, len(dims) + 1)) + [0])
+    Yr = np.transpose(np.reshape(images, (T, -1), order='F'))
```

Comment:
The code is doing this reshape and transpose operation here and then in `save_caiman_to_doric()`  it do other reshaping ([link](https://github.com/doriclenses/pyDoric/pull/36/files#diff-ac2c457f8f4d262f27bcfe124d6b97dc87a6623ac666f7b572d1a0e38b69d405R217)). I wonder if we could remove one of them. What do you think? @doriclenses

Reply:
It's possible it's redundant.. don't remember already. You can rework it and see if it works fine

Reply:
I'm gonna try to check if I can do something